includes/orphaned_objects.inc

```diff
@@ -0,0 +1,268 @@
+    ksort($rows);
+    $form['management_table'] = array(
+      '#type' => 'tableselect',
+      '#header' => array(t('Object')),
+      '#options' => $rows,
+      '#attributes' => array(),
+      '#empty' => t('No orphaned objects were found.'),
+    );
+    if (!empty($rows)) {
+      $form['submit_selected'] = array(
+        '#type' => 'submit',
+        '#name' => 'islandora-orphaned-objects-submit-selected',
+        '#validate' => array('islandora_orphaned_objects_manage_delete_selected_validate'),
+        '#submit' => array('islandora_orphaned_objects_manage_delete_submit'),
+        '#value' => t('Delete Selected'),
+      );
+      $form['submit_all'] = array(
+        '#type' => 'submit',
+        '#name' => 'islandora-orphaned-objects-submit-all',
+        '#submit' => array('islandora_orphaned_objects_manage_delete_submit'),
+        '#value' => t('Delete All'),
+      );
+    }
+  }
+  return $form;
+}
+/**
+ * Validation for the Islandora Orphaned Objects management form.
+ *
+ * @param array $form
+ *   An array representing a form within Drupal.
+ * @param array $form_state
+ *   An array containing the Drupal form state.
+ */
+function islandora_orphaned_objects_manage_delete_selected_validate($form, $form_state) {
+  $selected = array_filter($form_state['values']['management_table']);
+  if (empty($selected)) {
+    form_error($form['management_table'], t('At least one object must be selected to delete!'));
+  }
+}
+/**
+ * Submit handler for the delete buttons in the workflow management form.
+ *
+ * @param array $form
+ *   An array representing a form within Drupal.
+ * @param array $form_state
+ *   An array containing the Drupal form state.
+ */
+function islandora_orphaned_objects_manage_delete_submit(&$form, &$form_state) {
+  if ($form_state['triggering_element']['#name'] == 'islandora-orphaned-objects-submit-selected') {
+    $selected = array_keys(array_filter($form_state['values']['management_table']));
+  }
+  else {
+    $selected = array_keys($form_state['values']['management_table']);
+  }
+  $form_state['pids_to_delete'] = $selected;
+  // Rebuild to show the confirm form.
+  $form_state['rebuild'] = TRUE;
+  $form_state['show_confirm'] = TRUE;
+}
+/**
+ * Submit handler for the workflow management confirm form.
+ *
+ * @param array $form
+ *   An array representing a form within Drupal.
+ * @param array $form_state
+ *   An array containing the Drupal form state.
+ */
+function islandora_orphaned_objects_manage_confirm_submit($form, &$form_state) {
+  $batch = islandora_orphaned_objects_delete_create_batch($form_state['pids_to_delete']);
+  batch_set($batch);
+}
+
+/**
+ * Query for orphaned objects.
+ *
+ * @return array
+ *   An array containing the results of the orphaned objects queries.
+ */
+function islandora_orphaned_objects_get_orphaned_objects() {
+  $connection = islandora_get_tuque_connection();
+  // First query: get standard objects.
+  $object_query = <<<EOQ
+SELECT ?object ?collection ?title
```

Comment:
So, there is an issue with This SPARQL: it actually gets you every object that is connected via these few predicates to a non-existant object.  (and its missing options) BUT! The same object could still be connected to a valid, existant Collection and appear in this list  So it would not be orphan actually, just would have a missing parent.  Also, it could be part of a compound but have an unexistant collection listed in their parent. You need to have a second check probably against the original list. Means (idea, some prob has a better one): get the list of PID that have a deceased parent, now check if those actually have a "living" parent before allowing them to be deleted.

Reply:
I'm happy for such a check to be added to the query - but I honestly don't know the first thing about SPARQL. I got it from Phil Redmon in [this Google group thread](https://groups.google.com/d/topic/islandora/Z3eoEXp_2KQ/discussion).

I would be open to anyone who knows how to SPARQL making a pull against my branch to add this check. Anyone you can think to tag, @DiegoPino?

Reply:
@bondjimbond let me give that SPARQL query a try. Probably can be done in a single query but maybe a second one is safer. Would making a commit to your pull branch be a good idea for you? The only issue is I´m already reviewing your pull and my commit would become part of yours, so conflict of multi-verses..

Reply:
I'm happy to get a pull to my branch from you. Can always finish up the review and then ask for a final review from @rosiel when it's done prior to merge.

Reply:
@DiegoPino Have you written any modifications to that SPARQL query locally? I'm happy to take a PR to my branch whenever you have a chance to look at it again.